src/Google.Api.Gax/Platform.cs

```diff
@@ -112,10 +111,28 @@ public GcePlatformDetails(string metadataJson, string projectId, string instance
         public string InstanceId { get; }
 
         /// <summary>
-        /// The zone name where this GCE instance is running.
+        /// The zone where this GCE instance is running.
+        /// This is in the format <code>projects/&lt;project-number&gt;/zones/&lt;zone-name&gt;</code>.
         /// </summary>
         public string ZoneName { get; }
 
+        /// <summary>
+        /// The zone name where this GCE instance is running.
+        /// This is equivalent to the value of the <code>&lt;zone-name&gt;</code> part in
+        /// <see cref="ZoneName"/>
+        /// </summary>
+        public string Location
+        {
+            get
```

Comment:
I didn't want to receive a value for Location in the constructor because it is public and don't want to change it. If we overload the constructor to receive Location, we still need to do something when the current constructor is used, like maybe `return ZoneName`?
I think overloading the constructor might be overkill, so maybe accept any ZoneName value, if we can parse it we return the <zone_name> part else we return ZoneName as is.

Reply:
I think this is fine, but the comment for the `zoneName` constructor parameter should describe the now-expected format.

Reply:
OK